### PR TITLE
Add Rules,Remediation and Tests for SLES-15-040430

### PR DESCRIPTION
**Pull request by yarunachalam**
#### Description:

- Add Rules,Remediation and Tests for SLES-15-040430

#### Rationale:

- Add SLE15 stig to 'Disable GDM Unattended or Automatic Login'

### CODE REVIEW

**Review by carlosmmatos — CHANGES_REQUESTED** [review]
change unattend references to unattended for completeness/readability

**Comment by yarunachalam** [comment]
@carlosmmatos , I need more specific comment for unattend referernces.

**Comment by ggbecker** [comment]
> @carlosmmatos , I need more specific comment for unattend referernces.

He is mentioning that the word `unattend` should be replaced by the word `unattended`. I think all the files name have that wrong spelled word. So you just need to rename some files.

**Comment by carlosmmatos** [comment]
> > @carlosmmatos , I need more specific comment for unattend referernces.
> 
> He is mentioning that the word `unattend` should be replaced by the word `unattended`. I think all the files name have that wrong spelled word. So you just need to rename some files.

Correct, this is basically a spelling fix request. I believe you meant to use `unattended` rather than `unattend` (which is not a word). You have instances of this in both your rule names, as well as within the code.

**Comment by yarunachalam** [comment]
Rename complete.

**Review by carlosmmatos — APPROVED** [review]

File: linux_os/guide/system/software/gnome/gnome_login_screen/gnome_gdm_disable_unattended_automatic_login/rule.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+severity: high
+
+identifiers:
+    cce@sle15: CCE-85723-5
+
+references:
+    disa@sle15: CCI-000366
+    nist@sle15: CM-6(b),CM-6.1(iv)
+    srg@sle15: SRG-OS-000480-GPOS-00229
```
Comment (jan-cerny) [inline linux_os/guide/system/software/gnome/gnome_login_screen/gnome_gdm_disable_unattended_automatic_login/rule.yml:29]: CCIs and SRGs are product independent, see eg. https://github.com/ComplianceAsCode/content/pull/6895
```suggestion
    disa: CCI-000366
    nist@sle15: CM-6(b),CM-6.1(iv)
    srg: SRG-OS-000480-GPOS-00229
```

File: linux_os/guide/system/software/gnome/gnome_login_screen/gnome_gdm_disable_unattended_automatic_login/oval/shared.xml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<def-group>
+  <definition class="compliance" id="gnome_gdm_disable_unattended_automatic_login" version="2">
+    {{{ oval_metadata("Disable the GNOME Display Manager (GDM) ability to allow users to
+      automatically login.") }}}
+    <criteria operator="AND">
+      <criterion comment="Disable GDM Automatic Login" test_ref="test_disable_automatic_login" />
+      <criterion comment="Disable GDM Automatic Login" test_ref="test_disable_unattended_login" />
```
Comment (jan-cerny) [inline linux_os/guide/system/software/gnome/gnome_login_screen/gnome_gdm_disable_unattended_automatic_login/oval/shared.xml:7]: They both have the same comment which makes them more difficult to distinguish.
```suggestion
      <criterion comment="Disable GDM Password Less Login" test_ref="test_disable_unattended_login" />
```

File: linux_os/guide/system/software/gnome/gnome_login_screen/gnome_gdm_disable_unattended_automatic_login/rule.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+description: |-
+    The GNOME Display Manager (GDM) can allow users to automatically login without
+    user interaction or credentials or unattended login. User should always be required to authenticate themselves
+    to the system that they are authorized to use. To disable user ability to automatically
+    login to the system, set the <tt>DISPLAYMANAGER_AUTOLOGIN=""</tt>
+    or <tt>DISPLAYMANAGER_PASSWORD_LESS_LOGIN="no"</tt> in the
+    <tt>/etc/sysconfig/displaymanager</tt>. For example:
+    <pre>DISPLAYMANAGER_AUTOLOGIN=""
+         DISPLAYMANAGER_PASSWORD_LESS_LOGIN="no"</pre>
```
Comment (jan-cerny) [inline linux_os/guide/system/software/gnome/gnome_login_screen/gnome_gdm_disable_unattended_automatic_login/rule.yml:15]:
```suggestion
     DISPLAYMANAGER_PASSWORD_LESS_LOGIN="no"</pre>
```

File: linux_os/guide/system/software/gnome/gnome_login_screen/gnome_gdm_disable_unattended_automatic_login/rule.yml (hunk 1 above)
Comment (yarunachalam) [reply linux_os/guide/system/software/gnome/gnome_login_screen/gnome_gdm_disable_unattended_automatic_login/rule.yml]: fixed

File: linux_os/guide/system/software/gnome/gnome_login_screen/gnome_gdm_disable_unattended_automatic_login/oval/shared.xml (hunk 2 above)
Comment (yarunachalam) [reply linux_os/guide/system/software/gnome/gnome_login_screen/gnome_gdm_disable_unattended_automatic_login/oval/shared.xml]: Fixed

File: linux_os/guide/system/software/gnome/gnome_login_screen/gnome_gdm_disable_unattended_automatic_login/rule.yml (hunk 3 above)
Comment (yarunachalam) [reply linux_os/guide/system/software/gnome/gnome_login_screen/gnome_gdm_disable_unattended_automatic_login/rule.yml]: removed unwanted space

**Comment by brett060102** [comment]
@jan-cerny The requested changes have been made. WE do see that the ci/prow/e2e-aws-ocp4-cis-node tests failed, but those failures don't seem to be related to our changes.
 Any ideas?

**Comment by brett060102** [comment]
@jan-cerny @carlosmmatos Is there a reason that this has not been merged? Thanks

File: linux_os/guide/system/software/gnome/gnome_login_screen/gnome_gdm_disable_unattended_automatic_login/rule.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    system security.
+
+severity: high
+
+identifiers:
+    cce@sle15: CCE-85723-5
+
+references:
+    disa: CCI-000366
+    nist@sle15: CM-6(b),CM-6.1(iv)
```
Comment (vojtapolasek) [inline linux_os/guide/system/software/gnome/gnome_login_screen/gnome_gdm_disable_unattended_automatic_login/rule.yml:28]:
```suggestion
    nist: CM-6(b),CM-6.1(iv)
```

**Comment by vojtapolasek** [comment]
Just a nitpick, see the comment. I have a question. Do you think that these config changes make sense only together or they could be applied separately?

**Pull request merged**